### PR TITLE
Feature/dm/compress plugins

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    if (configuration.getBoolean(PROPERTY_PLUGIN_COMPRESSION_ENABLE).orElse(false)) {
+      Path pack200Path = FileUtils.getPack200FilePath(jarFile);
+      pack200(jarFile, pack200Path, pluginKey);
+      String hash = calculateMd5(pack200Path);
+      RemotePluginFile compressedPlugin = new RemotePluginFile(pack200Path.getFileName().toString(), hash);
+      compressedPlugins.put(pluginKey, compressedPlugin);
+    }
+  }
+
+  @CheckForNull
```
Comment (janos-ss) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java]: it's actually never null :)

File: server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  public Map<String, RemotePluginFile> getPlugins() {
+    return new HashMap<>(compressedPlugins);
+  }
+
+  private String calculateMd5(Path filePath) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java]: MAJOR Make "calculateMd5" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    try (InputStream fis = new BufferedInputStream(Files.newInputStream(filePath))) {
+      return DigestUtils.md5Hex(fis);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java]: CRITICAL Use a stronger hashing algorithm than MD5. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2070)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }
+
+  public Map<String, RemotePluginFile> getPlugins() {
+    return new HashMap<>(compressedPlugins);
+  }
+
+  private static String calculateMd5(Path filePath) {
+    try (InputStream fis = new BufferedInputStream(Files.newInputStream(filePath))) {
+      return DigestUtils.md5Hex(fis);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java]: CRITICAL Use a stronger hashing algorithm than MD5. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2070)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**